lib/record_store/provider/oracle_cloud_dns.rb

```diff
@@ -0,0 +1,195 @@
+      def update(id, record, zone)
+        record_fqdn = Record.ensure_ends_without_dot(record.fqdn)
+        record_rdata = retrieve_rdata_based_on_record_type(record)
+
+        # Retrieve all records that you want to keep before it's updated because it will overwrite
+        all_records = []
+        client.get_zone_records(zone, zone_version: zone_version(zone)).each do |response|
+          all_records += response.data.items
+        end
+
+        update_zone_record_items = all_records.map do |r|
+          OCI::Dns::Models::RecordDetails.new(
+            domain: r.domain,
+            ttl: r.ttl,
+            rtype: r.rtype,
+            recordHash: r.record_hash,
+            rdata: r.rdata,
+            rrsetVersion: r.rrset_version,
+            isProtected: true,
+          )
+        end
+
+        update_zone_record_items << OCI::Dns::Models::RecordDetails.new(
+          domain: record_fqdn,
+          ttl: record.ttl,
+          rtype: record.type,
+          rdata: record_rdata
+        )
+        update_zone_record_items.delete_if { |r| id == r.record_hash }
+
+        client.update_zone_records(
+          zone,
+          OCI::Dns::Models::UpdateZoneRecordsDetails.new(items: update_zone_record_items),
+        )
+      end
+
+      def secrets
+        super.fetch('oracle_cloud_dns')
+      end
+
+      def build_from_api(api_record)
+        fqdn = Record.ensure_ends_with_dot(api_record.domain)
+
+        record_type = api_record.rtype
+        return if record_type == 'SOA'
+
+        record = {
+          ttl: api_record.ttl,
+          fqdn: fqdn.downcase,
+          record_id: api_record.record_hash,
+        }
+        case record_type
+        when 'A', 'AAAA'
+          record[:address] = api_record.rdata
+        when 'ALIAS'
+          record[:alias] = api_record.rdata
+        when 'CAA'
+          flags, tag, value = api_record.rdata.split
+          record[:flags] = flags.to_i
+          record[:tag] = tag
+          record[:value] = Record.unquote(value)
+
+        when 'CNAME'
+          record[:cname] = api_record.rdata
+        when 'MX'
+          preference, exchange = api_record.rdata.split
+          record[:preference] = preference.to_i
+          record[:exchange] = exchange
+        when 'NS'
+          record[:nsdname] = api_record.rdata
+        when 'SPF', 'TXT'
+          record.merge!(txtdata: Record.unquote(api_record.rdata))
+        when 'SRV'
+          priority, weight, port, host = api_record.rdata.split(' ')
+          record[:priority] = priority.to_i
+          record[:weight] = weight.to_i
+          record[:port] = port.to_i
+          record[:target] = Record.ensure_ends_with_dot(host)
+        else
+          raise NameError, "Unsupported record type: #{record_type}"
+        end
+        Record.const_get(record_type).new(record)
+      end
+
+      def retrieve_rdata_based_on_record_type(record)
+        record.rdata_txt
```

Comment:
Wait, if we have the `record.rdata_txt` value, why do we need to call the longest-name-method `retrieve_rdata_based_on_record_type(record)` instead of directly calling `record.rdata_txt` where we need it ?

Reply:
You're right! 👍